zebra/zapi_msg.c

```diff
@@ -1751,7 +1751,6 @@ static bool zapi_read_nexthops(struct zserv *client, struct prefix *p,
 
 		/* MPLS labels for BGP-LU or Segment Routing */
 		if (CHECK_FLAG(api_nh->flags, ZAPI_NEXTHOP_FLAG_LABEL)
-		    && api_nh->type != NEXTHOP_TYPE_IFINDEX
```

Comment:
Hmm, no, I don't think this is a problem. We have to have a destination for a labelled nexthop: where would the labelled packet go, if we just have an interface?

Reply:
would not that kind of config work with /32 routes ?
```
ip route 10.125.0.2/32 eth0 label 54
interface eth0
ip address 10.125.0.1/32
exit
```
other than that, I saw (but did not find any documentation on that about cisco command:
```
interface eth0
ip address A.B.C.D label <>
!
```
maybe this has more to do with ip addrlabel command than real command. I did not check about that.

Reply:
Have you actually tried that?

I don't know what cisco is doing there either - just commenting on what zebra is trying to enforce.

Reply:
I will investigate opnion of kernel experts on what is happening if link layer for a given ip address is not available.
Because on linux that kind of config is authorised.

Reply:
you seem to be saying that it should be possible to use an interface if that interface happens to have a /32 (or /128, I guess?). I think it's worth knowing whether zebra can correctly deal with that - does zebra install something via netlink that actually works in that case? If not, then there's more to do to make it work (or the zapi validation is correct and should not be changed).